packages/runner/src/run.ts

```diff
@@ -218,17 +228,22 @@ export async function runTest(test: Test, runner: VitestRunner) {
   updateTask(test, runner)
 }
 
+const processedError = new WeakSet()
 function failTask(result: TaskResult, err: unknown) {
   result.state = 'fail'
-  const errors = Array.isArray(err)
-    ? err
-    : [err]
-  for (const e of errors) {
-    const error = processError(e)
-    result.error ??= error
-    result.errors ??= []
-    result.errors.push(error)
-  }
+  const errors = Array.isArray(err) ? err : [err]
+  // errors maybe push in expect.soft, so we need to process them
```

Comment:
Can't you just process them in `expect.soft`? I would suggest moving `serializeError` to `vitest/utils` and import it from there